core/src/test/java/se/tre/freki/storage/MemoryStore.java

```diff
@@ -110,11 +110,11 @@ public void close() {
 
   @Nonnull
   @Override
-  public ListenableFuture<LabelMeta> getMeta(final LabelId uid,
+  public ListenableFuture<Optional<LabelMeta>> getMeta(final LabelId uid,
                                              final LabelType type) {
     final String qualifier = type.toString().toLowerCase() + "_meta";
     final LabelMeta meta = labelMetas.get(uid, qualifier);
-    return Futures.immediateFuture(meta);
+    return Futures.immediateFuture(Optional.of(meta));
```

Comment:
Using `Optional.of()` here will fail since it throws an exception when `meta` is null. Can you make sure there is a test that checks this?